src/badguy/dispenser.cpp

```diff
@@ -155,15 +154,14 @@ Dispenser::draw(DrawingContext& context)
 void
 Dispenser::activate()
 {
-  if (m_broken){
+  if (m_broken) {
     return;
   }
-  if (m_autotarget && !m_swivel){ // auto cannon sprite might be wrong
-    auto* player = get_nearest_player();
-    if (player) {
+
+  auto* player = get_nearest_player();
+  if (m_autotarget && !m_swivel && player) { // auto cannon sprite might be wrong
       m_dir = (player->get_pos().x > get_pos().x) ? Direction::RIGHT : Direction::LEFT;
       m_sprite->set_action(m_dir == Direction::LEFT ? "working-left" : "working-right");
-    }
   }
   m_dispense_timer.start(m_cycle, true);
```

Comment:
`get_nearest_player()` is expected to increase in computational cost with multiplayer, so we should avoid calling it when unnecessary.

The if's that could go before that call should be placed before to avoid impacting performance